mirgecom/gas_model.py

```diff
@@ -388,50 +390,50 @@ def make_fluid_state(cv, gas_model,
         # ~~~ we need to squeeze wall_model in gas_model because this is easily
         # accessible everywhere in the code
 
-        tau = gas_model.wall.decomposition_progress(material_densities)
-        epsilon = gas_model.wall.void_fraction(tau)
-        temperature = gas_model.wall.get_temperature(
-            cv=cv, material_densities=material_densities,
-            tseed=temperature_seed, tau=tau, gas_model=gas_model)
+        tau = gas_model.decomposition_progress(material_densities)
+        wv = PorousWallVars(
+            material_densities=material_densities,
+            tau=tau,
+            density=gas_model.solid_density(material_densities),
+            void_fraction=gas_model.wall_model.void_fraction(tau),
+            emissivity=gas_model.wall_model.emissivity(tau),
+            permeability=gas_model.wall_model.permeability(tau),
+            tortuosity=gas_model.wall_model.tortuosity(tau)
+        )
 
-        pressure = 1.0/epsilon*gas_model.eos.pressure(cv=cv, temperature=temperature)
+        # FIXME is there a way to pass the argument to "niter"?
+        temperature = gas_model.get_temperature(cv=cv, wv=wv,
+            tseed=temperature_seed, niter=3)
+
+        pressure = gas_model.get_pressure(cv, wv, temperature)
 
         if limiter_func:
-            cv = limiter_func(cv=cv, wv=material_densities, pressure=pressure,
-                              temperature=temperature, epsilon=epsilon,
-                              dd=limiter_dd)
+            cv = limiter_func(cv=cv, wv=wv, pressure=pressure,
+                              temperature=temperature, dd=limiter_dd)
 
-        dv = PorousFlowDependentVars(
+        dv = MixtureDependentVars(
             temperature=temperature,
             pressure=pressure,
             speed_of_sound=gas_model.eos.sound_speed(cv, temperature),
             smoothness_mu=smoothness_mu,
             smoothness_kappa=smoothness_kappa,
             smoothness_beta=smoothness_beta,
             species_enthalpies=gas_model.eos.species_enthalpies(cv, temperature),
-            material_densities=material_densities
         )
 
-        # ~~~ Modify transport vars to include solid effects
-        gas_tv = gas_model.transport.transport_vars(cv=cv, dv=dv, eos=gas_model.eos)
-
-        tv = GasTransportVars(
-            bulk_viscosity=(
-                gas_tv.bulk_viscosity),
-            viscosity=gas_model.wall.viscosity(
-                temperature, tau, gas_tv),
-            thermal_conductivity=gas_model.wall.thermal_conductivity(
-                cv, material_densities, temperature, tau, gas_tv),
-            species_diffusivity=gas_model.wall.species_diffusivity(
-                temperature, tau, gas_tv),
-        )
+        # FIXME I have to pass "gas_model" but this class is internal to "gas_model"
+        # Seems dumb to me but I dont know to access the other classes...
+        tv = gas_model.transport.transport_vars(cv, dv, wv, gas_model)
```

Comment:
I (tried to) follow the same structure of the fluid-only: EOS handles the physical definition of everything, and the model is (kinda of) a wrapper/encompassing class. Additional functions are required since we have to combine both solid and gas (for instance, the temperature). The `model` allows us to access everything, while its sub-classes are related to the different aspects (gas, wall, transport).

Since they are all wrapped together, it should be possible to access the mother/sibling classes without explicitly giving it as an argument, but I couldn't make it work.

Reply:
After taking a closer look: you're just passing the `gas_model` in to get access to the EOS and wall models, right? Could you pass those into the transport model's `__init__` and store them instead?